pkg/specgen/generate/kube/kube.go

```diff
@@ -1128,14 +1128,14 @@ func getContainerResources(container v1.Container) (v1.ResourceRequirements, err
 
 // getPodPorts converts a slice of kube container descriptions to an
 // array of portmapping
-func getPodPorts(containers []v1.Container) []types.PortMapping {
+func getPodPorts(containers []v1.Container, publishAll bool) []types.PortMapping {
 	var infraPorts []types.PortMapping
 	for _, container := range containers {
 		for _, p := range container.Ports {
 			if p.HostPort != 0 && p.ContainerPort == 0 {
 				p.ContainerPort = p.HostPort
 			}
-			if p.HostPort == 0 && p.ContainerPort != 0 {
+			if p.HostPort == 0 && p.ContainerPort != 0 && publishAll {
 				p.HostPort = p.ContainerPort
```

Comment:
Is this correct for K8S? From `podman run`/`docker run` with `--public-all`, I'd expect a random HostPort to be assigned in this case, not setting HostPort to ContainerPort

Reply:
@umohnani8 @ygalblum PTAL

Reply:
I agree that it should have been a random port. But, the question here is backward compatibility.
The new flag breaks the current behavior, and we agreed that it is OK to do so. However, with the proposed change, users can overcome this break by adding this flag. Assigning random ports will break the behavior even further.

Reply:
> I will need a decisive decision by the maintainers

@vrothberg @rhatdan @mheon @umohnani8 what do we decide here?  

I think we should leave the old behavior to allow a simple fix to the backward compatibility break.

Reply:
I agree with @ygalblum.  The goal of this PR is to stop publishing all ports by default with an easy way to get back to the previous behavior if needed.  Picking random ports would break the capability of restoring the previous behavior. :+1: